frontend/src/pages/review-destruction-list/ReviewDestructionList.tsx

```diff
@@ -0,0 +1,233 @@
+      type: "radio", // TODO: Radio not implemented yet (I think?)
+      options: [
+        { label: "Vernietigen uitstellen", value: "true" },
+        { label: "Niet vernietigen", value: "false" },
+        // If we check "true" then we should show a date field to select the date when the destruction should be done.
+        // If we check "false" then we should show a (combobox) text/select field to enter the reason why the destruction should not be done. (which is required)
+      ],
+      required: true,
+    },
+    ...(formData?.archief_informatie === "false"
+      ? [
+          {
+            label: "Reden",
+            name: "reason",
+            type: "select",
+            options: [
+              { label: "Nog rechtzaak lopend", value: "rechtzaak" },
+              {
+                label: "Geen toestemming van de eigenaar",
+                value: "toestemming",
+              },
+              {
+                label: "Andere reden",
+                value: "other",
+              },
+            ], // TODO: From backend these options?
+            required: true,
+          },
+          ...(formData?.reason === "other"
+            ? [
+                {
+                  label: "Andere reden",
+                  name: "other_reason",
+                  type: "text",
+                  required: true,
+                },
+              ]
+            : []),
+        ]
+      : [
+          {
+            label: "", // Design has no label for this field.
+            name: "date",
+            type: "date",
+            required: true,
+          },
+        ]),
+  ];
+
+  /**
+   * Gets called when the form is submitted.
+   */
+  const onSubmitForm = async (event: FormEvent, data: SerializedFormData) => {
+    const zaakSelection = await getZaakSelection(destructionListReviewKey);
+    const zaakUrls = Object.entries(zaakSelection)
+      .filter(([, selected]) => selected)
+      .map(([url]) => url);
+    const formData = data as unknown as FormDataState;
+
+    submit(
+      {
+        ...formData,
+        zaken: zaakUrls,
+      },
+      { method: "POST", encType: "application/json" },
+    );
+    setModalDataState({ open: false });
+  };
+
+  return (
+    <>
+      <Modal
+        title={modalDataState.title}
+        open={modalDataState.open}
+        size="m"
+        onClose={() => setModalDataState({ open: false })}
+      >
+        <Body>
+          <Form
+            fields={modalFormFields}
+            onSubmit={onSubmitForm}
+            validate={(values) =>
+              setFormData(values as unknown as FormDataState)
+            }
+            validateOnChange={true}
+          />
+        </Body>
+      </Modal>
+      <DestructionList
+        zaken={zaken}
+        onSubmitSelection={onSubmitSelection}
+        selectedZaken={selectedZaken}
+        zaaktypeChoices={zaaktypeChoices}
+        destructionListCreateKey={destructionListReviewKey}
+        title="Vernietigingslijst reviewen"
```

Comment:
This is not the only label that should be modifier for this view, the review button now has a label "Vernietigingslijst aanmaken" (probably inherited from the base component). 

The form action should not be a default and should be provided by the implementation. 

Motivation: Explicitly specifying actions is a safer bet here, also nice labels matter :)